PowerEditor/src/Utf8_16.cpp

```diff
@@ -335,38 +353,50 @@ bool Utf8_16_Write::writeFile(const void* p, size_t _size)
 		case uni7Bit:
         case uni8Bit:
         case uniCookie:
-        case uniUTF8: {
+        case uniUTF8:
+		{
             // Normal write
 			if (m_pFile->write(p, _size))
 				isOK = true;
-            break;
+
         }
+		break;
+
         case uni16BE_NoBOM:
         case uni16LE_NoBOM:
         case uni16BE:
-        case uni16LE: {
+        case uni16LE:
+		{
 			static const unsigned int bufSize = 64*1024;
 			utf16* buf = new utf16[bufSize];
             
             Utf8_Iter iter8;
             iter8.set(static_cast<const ubyte*>(p), _size, m_eEncoding);
 
 			unsigned int bufIndex = 0;
-			while (iter8) {
+			while (iter8)
+			{
 				++iter8;
 				while ((bufIndex < bufSize) && iter8.canGet())
 					iter8.get(&buf [bufIndex++]);
 
-				if (bufIndex == bufSize || !iter8) {
-					if (!m_pFile->write(buf, bufIndex*sizeof(utf16))) return 0;
+				if (bufIndex == bufSize || !iter8)
+				{
+					if (!m_pFile->write(buf, bufIndex * sizeof(utf16)))
+					{
+						delete[] buf;

```

Comment:
Seems this is the relevant change. Likely usage of a smartpointer could help to avoid such missing deletes.